theano/sandbox/cuda/opt.py

```diff
@@ -1068,6 +1070,8 @@ def local_gpu_incsubtensor(node):
             incsubt = host_output.owner.op
             x, y = host_output.owner.inputs[0:2]
             coords = host_output.owner.inputs[2:]
+            if x.dtype != "float32" or y.dtype != "float32":
+                return
```

Comment:
I don't think that is right. If the value of the increment is not float32, it should be casted, as this is what happens inside the "set_subtensor" anyway.
This is what the comment below is about, btw.